### PR TITLE
Fix spelling in error messages

### PR DESCRIPTION
**Pull request by hgzimmerman**
Implements #57 .

As of the first commit, this only makes spelling changes to the error messages, and does not attempt to change the names of the variants.
I'm happy to discuss how to change the variant names in a manner that causes an acceptable level of breakage.

- [x] cargo fmt has been run
- [x] cargo test has been run and passes
- [x] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
@hgzimmerman The current content in latest git master is currently undergoing an alpha series as we are making a number of major changes. So feel free to change/fix the other spelling errors in the names of the variants. IE breaking api changes are allowed today.

**Comment by hgzimmerman** [comment]
I think I've done all that I intend to do with this PR. Waiting on final approval or feedback.

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
@hgzimmerman This looks excellent, thank you! As mentioned, there are some upcoming breaking changes still for 0.3, so there is still time for some other improvements. I will likely do the bulk of these in the next week or two.

**Pull request merged**
